### PR TITLE
Add tests for java_tools.zip contents.

### PR DESCRIPTION
**Pull request by iirina**

### CODE REVIEW

**Comment by lberki** [comment]
I'm afraid I won't be able to do this review this week -- I'm travelling and I'm afraid I have to prioritize other things over this for the next few days.

File: src/test/shell/bazel/bazel_java_tools_test.sh [hunk 1]
```diff
@@ -0,0 +1,152 @@
+#!/bin/bash
+#
+# Copyright 2016 The Bazel Authors. All rights reserved.
```
Comment (meisterT) [inline src/test/shell/bazel/bazel_java_tools_test.sh]: nit: it's 2019 already ;-)

File: src/test/shell/bazel/bazel_java_tools_test.sh [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+source "$(rlocation "io_bazel/src/test/shell/integration_test_setup.sh")" \
+  || { echo "integration_test_setup.sh not found!" >&2; exit 1; }
+
+# `uname` returns the current platform, e.g "MSYS_NT-10.0" or "Linux".
+# `tr` converts all upper case letters to lower case.
+# `case` matches the result if the `uname | tr` expression to string prefixes
+# that use the same wildcards as names do in Bash, i.e. "msys*" matches strings
+# starting with "msys", and "*" matches everything (it's the default case).
+case "$(uname -s | tr [:upper:] [:lower:])" in
```
Comment (meisterT) [inline src/test/shell/bazel/bazel_java_tools_test.sh]: It feels wrong to define this in every test and not somewhere centrally.
(This is unrelated to your change, so cc-ing @laszlocsomor.)

File: src/test/shell/bazel/bazel_java_tools_test.sh [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+fi
+
+function expect_path_in_java_tools() {
+  path="$1"; shift
+
+  count=$(zipinfo -1 $(rlocation io_bazel/src/java_tools.zip) | grep -c "$path")
+  [[ "$count" -gt 0 ]] || fail "Path $path not found in java_tools"
+}
+
+function test_java_tools_has_ijar() {
```
Comment (meisterT) [inline src/test/shell/bazel/bazel_java_tools_test.sh]: I personally would move all of these into one single test. Feel free to leave it as is if you prefer individual tests.

File: src/test/shell/bazel/bazel_java_tools_test.sh (hunk 2 above)
Comment (laszlocsomor) [reply src/test/shell/bazel/bazel_java_tools_test.sh]: Yes it does, but I have no better idea.
See https://github.com/bazelbuild/bazel/blob/c0ac80e0e56e923e1c54c973c65cec26b8a52081/tools/bash/runfiles/runfiles.bash#L34-L36

Comment (meisterT) [reply src/test/shell/bazel/bazel_java_tools_test.sh]: I'm not talking about the runfiles setup and rlocation, I am aware of that part.

I was referring to the `is_windows` related code snippet.

Comment (laszlocsomor) [reply src/test/shell/bazel/bazel_java_tools_test.sh]: Yes, that we could move to a central file.

File: src/test/shell/bazel/bazel_java_tools_test.sh (hunk 3 above)
Comment (iirina) [reply src/test/shell/bazel/bazel_java_tools_test.sh]: I find it easier to spot the failures in individual tests.
